### PR TITLE
[feature] support fluentd_pid_dir and fluentd_pid_file

### LINKED ISSUE #38: the role does not manage pid directory and pid path
although the packages handle these without issues at the moment, it would be nice to have a variable, `fluentd_pid_file`. the variable can be used to send a signal to the daemon. using a variable is better than hard-coded path to PID file.

an example use case is, when you want to send `USR1` to notify the daemon to flush events in its buffers. Another is, when its log file has been rotated. see [Fluentd’s Signal Handling](http://docs.fluentd.org/v0.12/articles/signals).

### PR DESCRIPTION
**Pull request by trombik**
fixes #38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +68,8 @@ Note that although the role creates `fluentd_log_dir`, you need to configure
 | `__fluentd_gem_bin` | `/usr/sbin/td-agent-gem` |
 | `__fluentd_unix_pipe_dir` | `/var/tmp/fluentd` |
 | `__fluentd_flags` | `""` |
+| `__fluentd_pid_dir` | `/var/run/fluentd` |
```
Comment (mitsururike) [inline README.md]: /fluentd -> /td-agent

File: README.md [hunk 2]
```diff
@@ -66,6 +68,8 @@ Note that although the role creates `fluentd_log_dir`, you need to configure
...
+| `__fluentd_pid_file` | `{{ fluentd_pid_dir }}/fluentd.pid` |
```
Comment (mitsururike) [inline README.md]: /fluentd -> /td-agent

File: README.md [hunk 3]
```diff
@@ -96,6 +102,9 @@ Note that although the role creates `fluentd_log_dir`, you need to configure
 | `__fluentd_gem_bin` | `/usr/sbin/td-agent-gem` |
 | `__fluentd_unix_pipe_dir` | `/var/tmp/fluentd` |
 | `__fluentd_flags` | `""` |
+| `__fluentd_pid_dir` | `/var/run/td-agent` |
+| `__fluentd_pid_file` | `{{ fluentd_pid_dir }}/fluentd.pid` |
```
Comment (mitsururike) [inline README.md]: /fluentd -> /td-agent

File: tasks/main.yml [hunk 4]
```diff
@@ -143,3 +143,30 @@
     name: "{{ fluentd_service_name }}"
     enabled: true
     state: started
+
+- name: Check if pid directory exists
```
Comment (mitsururike) [inline tasks/main.yml:147]: file module with state 'directory' is simpler.

File: tasks/main.yml [hunk 5]
```diff
@@ -143,3 +143,30 @@
...
+  when: not ansible_check_mode
+
+- name: Assert pid directory exists
+  assert:
+    that:
+      - register_fluentd_stat_pid_dir.stat.exists
+      - register_fluentd_stat_pid_dir.stat.isdir
+  when: not ansible_check_mode
+
+- name: Check if pid file exists
```
Comment (mitsururike) [inline tasks/main.yml:168]: some explanation should be written why fluentd fails to create pid file.
restarting fluentd does not solve the problem?

Comment (trombik) [reply tasks/main.yml:168]: as commented in cb74d94cd09b1a671f1b7780fd7f4036b423927d, they are a test in `ansible` play. `serverspec` helps role development a lot but does not when used in a project. we should have [more tests in roles](http://docs.ansible.com/ansible/test_strategies.html) as well.

**Review by mitsururike — APPROVED** [review]

**Pull request merged**